data/crac-io/crac-io-json/src/main/java/com/farao_community/farao/data/crac_io_json/serializers/CounterTradeRangeActionSerializer.java

```diff
@@ -0,0 +1,48 @@
+package com.farao_community.farao.data.crac_io_json.serializers;
+
+import com.farao_community.farao.data.crac_api.range.StandardRange;
+import com.farao_community.farao.data.crac_api.range_action.CounterTradeRangeAction;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static com.farao_community.farao.data.crac_io_json.JsonSerializationConstants.*;
+
+/**
+ * @author Gabriel Plante {@literal <gabriel.plante_externe at rte-france.com>}
+ */
+public class CounterTradeRangeActionSerializer extends AbstractJsonSerializer<CounterTradeRangeAction> {
+
+    @Override
+    public void serialize(CounterTradeRangeAction value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
```

Comment:
Code duplication.. Could this be avoided ?

Reply:
maybe you can add a StandardRangeActionSerializationCommons to put common code. To have a readable order of elements in the json file, you can divide the common code into two parts : 
- StandardRangeActionSerializationCommons.serializeMainAttributes : serializes ID, name, initial setpoint.. all the small attributes
- StandardRangeActionSerializationCommons.serializeOtherAttributes : seralizes difficult-to-read attributes, such as ranges and usage rules
Then in your specific serializer, call the serializers in this order : 
1. StandardRangeActionSerializationCommons.serializeMainAttributes
2. SpecificSerializer -> serialize its specific attributes (in this example, exporting & importing countries)
3. StandardRangeActionSerializationCommons.serializeOtherAttributes

Of course, we could merge StandardRangeActionSerializationCommons.serializeMainAttributes & StandardRangeActionSerializationCommons.serializeOtherAttributes, but that would make the output json less readable.
PS : I just noticed that actually we serialize usage rules in the middle of the RA, I think it would be better to have them at the end